### PR TITLE
LPS-114169 Use Liferay Product Key

### PR DESCRIPTION
**Pull request by lawrence-lee**

### CODE REVIEW

File: modules/sdk/project-templates/project-templates-workspace/src/main/resources/archetype-resources/gradle.properties [hunk 1]
```diff
@@ -98,7 +92,25 @@
...
 
+
+    #
+    # Set the liferay.workspace.product.key to set the app.server.tomcat.version,
+    # liferay.workspace.bundle.url, liferay.workspace.docker.image.liferay, and
+    # liferay.workspace.target.platform.version that matches your Liferay Product
+    # Version. To override each of these settings, set them individually.
+
+#if (${liferayVersion.startsWith("7.0")})
+    liferay.workspace.product.key=portal-7.0-ga7
```
Comment (gamerson) [inline modules/sdk/project-templates/project-templates-workspace/src/main/resources/archetype-resources/gradle.properties]: The name of the key is `liferay.workspace.product`

File: modules/sdk/project-templates/project-templates-workspace/src/main/resources/archetype-resources/gradle.properties [hunk 2]
```diff
@@ -10,36 +10,29 @@
...
-#if (${liferayVersion.startsWith("7.0")})
-    liferay.workspace.bundle.url=https://releases-cdn.liferay.com/portal/7.0.6-ga7/liferay-ce-portal-tomcat-7.0-ga7-20180507111753223.zip
-#elseif (${liferayVersion.startsWith("7.1")})
-    liferay.workspace.bundle.url=https://releases-cdn.liferay.com/portal/7.1.3-ga4/liferay-ce-portal-tomcat-7.1.3-ga4-20190508171117552.tar.gz
-#elseif (${liferayVersion.startsWith("7.2")})
-    liferay.workspace.bundle.url=https://releases-cdn.liferay.com/portal/7.2.1-ga2/liferay-ce-portal-tomcat-7.2.1-ga2-20191111141448326.tar.gz
-#elseif (${liferayVersion.startsWith("7.3")})
-    liferay.workspace.bundle.url=https://releases-cdn.liferay.com/portal/7.3.2-ga3/liferay-ce-portal-tomcat-7.3.2-ga3-20200519164024819.tar.gz
-#end
+    #liferay.workspace.bundle.url=https://releases-cdn.liferay.com/portal/7.0.6-ga7/liferay-ce-portal-tomcat-7.0-ga7-20180507111753223.zip
```
Comment (gamerson) [inline modules/sdk/project-templates/project-templates-workspace/src/main/resources/archetype-resources/gradle.properties]: lets just show the latest GA, so 7.3.2-ga3.  No need to list all the previous GAs.  they are publicly available here:

https://releases-cdn.liferay.com/portal/

File: modules/sdk/project-templates/project-templates-workspace/src/main/resources/archetype-resources/gradle.properties [hunk 3]
```diff
@@ -10,36 +10,29 @@
     #liferay.workspace.bundle.cache.dir=~/.liferay/bundles
 
     #
+    # Legacy Feature (See liferay.workspace.product.key)
```
Comment (gamerson) [inline modules/sdk/project-templates/project-templates-workspace/src/main/resources/archetype-resources/gradle.properties]: Lets not call it Legacy, as we still want to support it going forward.

Instead call it `Override bundle url` or something, so they know this section is for overriding the `liferay.workspace.product`

**Comment by lawrence-lee** [comment]
@gamerson Can you review this first? I need to bump project-templates tokens in order to test this.

File: modules/sdk/project-templates/project-templates-workspace/src/main/resources/archetype-resources/gradle.properties [hunk 4]
```diff
@@ -10,36 +10,23 @@
     #liferay.workspace.bundle.cache.dir=~/.liferay/bundles
 
     #
+    # Override bundle url
+    # Set this property to override the default setting in liferay.workspace.product
```
Comment (gamerson) [inline modules/sdk/project-templates/project-templates-workspace/src/main/resources/archetype-resources/gradle.properties]:
```suggestion
    # Set this property to override the default setting provided by liferay.workspace.product
```

**Comment by lawrence-lee** [comment]
ci:test:project-templates

**Comment by lawrence-lee** [comment]
ci:test:sf

**Comment by lawrence-lee** [comment]
ci:test:sf

**Comment by lawrence-lee** [comment]
ci:test:project-templates

**Comment by lawrence-lee** [comment]
ci:stop

**Comment by lawrence-lee** [comment]
ci:test:sf

**Comment by lawrence-lee** [comment]
ci:test:project-templates

**Comment by gamerson** [comment]
ci:test:project-templates

File: modules/sdk/project-templates/project-templates-api/src/main/resources/archetype-resources/pom.xml [hunk 5]
```diff
@@ -9,49 +9,62 @@
 	<groupId>${groupId}</groupId>
 	<artifactId>${artifactId}</artifactId>
 	<version>${version}</version>
-	<properties>
+	<profiles>
+		<profile>
+
+			<!-- Enable this profile is standalone -->
```
Comment (gamerson) [inline modules/sdk/project-templates/project-templates-api/src/main/resources/archetype-resources/pom.xml]:
```suggestion
			<!-- Enable this profile in standalone -->
```

File: modules/sdk/project-templates/project-templates-api/src/main/resources/archetype-resources/pom.xml [hunk 6]
```diff
@@ -9,49 +9,62 @@
 	<groupId>${groupId}</groupId>
 	<artifactId>${artifactId}</artifactId>
 	<version>${version}</version>
-	<properties>
+	<profiles>
+		<profile>
+
```
Comment (gamerson) [inline modules/sdk/project-templates/project-templates-api/src/main/resources/archetype-resources/pom.xml:14]: lets remove this whitespace

File: modules/sdk/project-templates/project-templates-workspace/src/main/resources/archetype-resources/gradle.properties [hunk 7]
```diff
@@ -10,36 +10,23 @@
...
     #
-    # Set the URL pointing to the bundle Zip to download. If the URL points to a
-    # DXP bundle (e.g., https://api.liferay.com/...), The default value is the
-    # URL for Liferay Portal CE 7.0 GA7, Liferay Portal CE 7.1 GA4, Liferay
-    # Portal CE 7.2 GA2, or Liferay Portal CE 7.3 GA1, depending on the portal
-    # version the workspace is targeting.
+    # Set this property to override the default setting provided by
+    # liferay.workspace.product. Set the URL pointing to the bundle Zip to
+    # download. If the URL points to a DXP bundle (e.g.,
+    # https://api.liferay.com/...)
```
Comment (gamerson) [inline modules/sdk/project-templates/project-templates-workspace/src/main/resources/archetype-resources/gradle.properties]: can we include a URL to show where to find all of the portal versions?

**Comment by lawrence-lee** [comment]
ci:test:sf

**Comment by lawrence-lee** [comment]
ci:test:project-templates

**Comment by lawrence-lee** [comment]
ci:stop

**Comment by lawrence-lee** [comment]
ci:stop

**Comment by lawrence-lee** [comment]
ci:test:sf

**Comment by lawrence-lee** [comment]
ci:test:project-templates

**Comment by lawrence-lee** [comment]
Sent to https://github.com/petershin/liferay-portal/pull/1037

**Pull request closed without merging**